### PR TITLE
Ensure prop dictionary only contains strings

### PR DESCRIPTION
**Pull request by spike77453**
https://github.com/389ds/389-ds-base/blob/829ea4113fee15f9bf5779734f9d2ddf0281e49b/src/lib389/lib389/__init__.py#L559

Since `self.ldapi_enabled` is a bytes object, `self.ldapi_enabled == 'on'` will be `False` even if `self.ldapi_enabled` is `b'on'`. Bytes objects need to be decoded before comparing them to strings.

### CODE REVIEW

**Comment by spike77453** [comment]
Thinking about this some more, it might be a good idea to make all values of the `args` dictionary stings, instead of some of them being bytes and some being strings. So maybe changing

https://github.com/389ds/389-ds-base/blob/829ea4113fee15f9bf5779734f9d2ddf0281e49b/src/lib389/lib389/__init__.py#L699

to `prop[key] = configentry.getValue(args_dse_keys[key]).decode()` is the more sane approach?

**Comment by Firstyear** [comment]
We have a wrapper for this that's python2/3 compaible for lol reasons. I think it's ensure_str or ensure_bytes which you can use to wrap the getValue.

**Comment by spike77453** [comment]
> We have a wrapper for this that's python2/3 compaible for lol reasons. I think it's ensure_str or ensure_bytes which you can use to wrap the getValue.

I've updated this PR accordingly.

**Review by progier389 — APPROVED** [review]
LGTM

**Comment by spike77453** [comment]
I there anything I can do or still need to do to get this merged?

**Comment by progier389** [comment]
Since you got the approval,  I think that you only have  to hit the "Squash and merge" button.

**Comment by Firstyear** [comment]
@progier389 They won't have commit rights to squash/merge themself, one of us need to do it.

**Comment by progier389** [comment]
Thank you @FirstYear for the information, so I merged it ...

**Pull request merged**
